src/devices/src/virtio/net/tap.rs

```diff
@@ -35,6 +36,40 @@ ioctl_iow_nr!(TUNSETIFF, TUNTAP, 202, ::std::os::raw::c_int);
 ioctl_iow_nr!(TUNSETOFFLOAD, TUNTAP, 208, ::std::os::raw::c_uint);
 ioctl_iow_nr!(TUNSETVNETHDRSZ, TUNTAP, 216, ::std::os::raw::c_int);
 
+struct IfReqBuilder(ifreq);
+
+impl IfReqBuilder {
+    fn new() -> Self {
+        Self(Default::default())
+    }
+
+    fn if_name(mut self, if_name: &[u8; IFACE_NAME_MAX_LEN]) -> Self {
+        // Since we don't call as_mut on the same union field more than once, this block is safe.
+        let ifrn_name = unsafe { self.0.ifr_ifrn.ifrn_name.as_mut() };
+        ifrn_name.copy_from_slice(if_name.as_ref());
+
+        self
+    }
+
+    fn flags(mut self, flags: i16) -> Self {
+        // Since we don't call as_mut on the same union field more than once, this block is safe.
+        let ifru_flags = unsafe { self.0.ifr_ifru.ifru_flags.as_mut() };
+        *ifru_flags = flags;
+
+        self
+    }
+
+    fn execute<F: AsRawFd>(mut self, socket: &F, ioctl: u64) -> Result<ifreq> {
```

Comment:
Would be better if we have this on Tap? It is not relevant to `ifreq` building.

Reply:
I was a bit unsure here. It's true `execute` is not part of the `ifreq` building process. But it's not really related to the tap either. But since it's more related to the `ifreq` than to the `tap`, I defined it here. Maybe it's a naming issue ? How about renaming it `build_and_execute` ? 

Another option would be to have a new class `IfReq` that holds a `ifreq` inside. And define `execute` there and `build` on `IfReqBuilder`. I think it would be more fair, but it would add some more code and right now we don't really need it.

WDYT ?

Reply:
It is a bit uncertain for me why we would need to execute anything, while building an `ifreq`. Then, why executing  means doing an ioctl on the socket. I was seeing it more like a method used for configuring a network device [1]. I would rather rename the method to something like `configure_netif`, which will end building an `ifreq`, or move it on the Tap as `configure`. WDYT?

[1] netdevice documentation: "Linux supports some standard ioctls to configure network devices. They can be used on any socket's file descriptor regardless of the family or type. Most of them pass an ifreq structure [...]".

Reply:
Talked with @serban300 offline. `execute` will have as an side effect issuing an `ifreq`, which stands for interface request. Resolving this.